src/components/Form/Location/Address.jsx

```diff
@@ -85,49 +71,12 @@ export default class Address extends ValidationElement {
     }
   }
 
-  updateStreet(values) {
-    this.update(
-      {
-        street: values.value
-      },
-      6000
-    )
-  }
-
-  updateStreet2(values) {
-    this.update(
-      {
-        street2: values.value
-      },
-      6000
-    )
-  }
-
-  updateCity(values) {
-    this.update(
-      {
-        city: values.value
-      },
-      6000
-    )
-  }
-
-  updateState(values) {
-    this.update(
-      {
-        state: values.value
-      },
-      6000
```

Comment:
I removed all of the delay values. I'm not sure what the reasoning behind them is, or why they used the magic number of 6000 ms

Reply:
Is it connected to giving folks an opportunity to finish typing before kicking off validation logic or the web service call to USPS validation API? I don't have anything at my finger tips, but I seem to recall seeing the usability studies where they mention they were tweaking these delay values.

Reply:
USPS validation is not enabled by default, here are some instructions:
https://github.com/18F/e-QIP-prototype/blob/develop/docs/usps.md

Reply:
got it! In that case, a long delay is probably fine. I'll try adding it in, and make a constant to represent the delay time if needed

Reply:
I added the delay back. and there was no different in how fast the modal appeared. I believe that behavior is caused by another call to update, and is a more reasonable 500ms